src/main/java/codeu/model/data/Activity.java

```diff
@@ -0,0 +1,76 @@
+package codeu.model.data;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.time.format.FormatStyle;
+import java.util.Locale;
+import java.util.UUID;
+import java.util.List;
+
+
+public class Activity implements Comparable<Activity> {
+	public final UUID id;
+	public final Instant creation;
+	public final String type;
+	public final List<String> attributes;
+
+	public Activity(UUID id, Instant creation, String type, List<String> attributes) {
```

Comment:
If you go with the OOP approach (as mentioned below) you could get rid of the List of attributes and just have named parameters that are specific to each sub class.  A list of attributes is a little funny in Java code because it assumes an ordering that must be the same between the class instantiator and the code that prints out the string.